File: python/scrabble-score/scrabble_score.py
# Human-readable score map.
SCORE_MAP = {
    1  : 'aeioulnrst',
    2  : 'dg',
    3  : 'bcmp',
    4  : 'fhvwy',
    5  : 'k',
    8  : 'jx',
    10 : 'qz'
}
# Transposed for easier scoring.
POINTS = { v: k for k, vs in SCORE_MAP.items() for v in vs }

def score(word):
    return sum(POINTS[l] for l in word.lower())

File: typescript/scrabble-score/scrabble-score.ts
const scoreMap: {[key: number]: string} = {
  1: "aeioulnrst",
  2: "dg",
  3: "bcmp",
  4: "fhvwy",
  5: "k",
  8: "jx",
  10: "qz",
};

let points: {[key: string]: number} = {};
for (let key in scoreMap) {
  if (scoreMap.hasOwnProperty(key)) {
    [...scoreMap[key]].forEach((i) => { points[i] = parseInt(key); });
  }
};

const score = (word: string): number => {
  if (!word) { return 0; }
  return [...word.toLowerCase()].reduce((acc, i) => acc + points[i], 0);
}

export default score;